test/e2e/Helper.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Test, Vm, stdJson} from "forge-std/Test.sol";
+import {IERC20} from "src/contracts/interfaces/IERC20.sol";
+
+import {GPv2AllowListAuthentication} from "src/contracts/GPv2AllowListAuthentication.sol";
+import {
+    GPv2Authentication,
+    GPv2Interaction,
+    GPv2Settlement,
+    GPv2Trade,
+    IERC20,
+    IVault
+} from "src/contracts/GPv2Settlement.sol";
+
+import {WETH9} from "./WETH9.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+import {SwapEncoder} from "test/libraries/encoders/SwapEncoder.sol";
+
+address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+interface IAuthorizer {
+    function grantRole(bytes32, address) external;
+}
+
+interface IERC20Mintable is IERC20 {
+    function mint(address, uint256) external;
+    function burn(uint256) external;
+}
+
+// solhint-disable func-name-mixedcase
+// solhint-disable max-states-count
+abstract contract Helper is Test {
+    using stdJson for string;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    address internal deployer;
+    address internal owner;
+    GPv2Settlement internal settlement;
+    bytes32 internal domainSeparator;
+    GPv2Authentication internal authenticator;
+    IVault internal vault;
+    GPv2AllowListAuthentication internal allowList;
+    GPv2AllowListAuthentication internal allowListImpl;
+    address vaultRelayer;
+    address balancerVaultAuthorizer;
+
+    SettlementEncoder.State internal encoder;
+    SwapEncoder.State internal swapEncoder;
+
+    address internal solver;
+    Vm.Wallet internal trader;
+
+    bool immutable isForked;
+    uint256 forkId;
+
+    WETH9 weth;
+
+    bytes32 constant SALT = "Mattresses in Berlin!";
+
+    constructor(bool _isForked) {
+        isForked = _isForked;
+    }
+
+    function setUp() public virtual {
+        if (isForked) {
+            uint256 blockNumber = vm.envUint("FORK_BLOCK_NUMBER");
+            string memory forkUrl = vm.envString("FORK_URL");
+            forkId = vm.createSelectFork(forkUrl, blockNumber);
+            weth = WETH9(payable(WETH));
```

Comment:
I'd suggest that for forked tests, there should be an in-code opinionated RPC automatically set in the event that the environment variable `FORK_URL` is not set.